### PR TITLE
[book] fixes typo about redirect status codes in the controller chapter.

### PR DESCRIPTION
**Pull request by hhamon**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.6+ |
| Fixed tickets | ~ |

### CODE REVIEW

File: book/controller.rst [hunk 1]
```diff
@@ -438,7 +438,7 @@ If you want to redirect the user to another page, use the ``redirectToRoute()``
         return $this->redirectToRoute('homepage');
 
         // redirectToRoute is equivalent to using redirect() and generateUrl() together:
-        // return $this->redirect($this->generateUrl('homepage'), 301);
```
Comment (javiereguiluz) [inline book/controller.rst]: I'm not sure this is a typo. The default value of the second argument of `redirect` is `302`:

```php
    protected function redirect($url, $status = 302)
    {
        return new RedirectResponse($url, $status);
    }
```

Comment (hhamon) [inline book/controller.rst]: By default, both `redirect()` and `redirectToRoute()` do a `302` not a `301`, which must be explicitely pass to the function.

Comment (hhamon) [inline book/controller.rst]:
```php
class Controller
{
    /**
     * Returns a RedirectResponse to the given URL.
     *
     * @param string $url    The URL to redirect to
     * @param int    $status The status code to use for the Response
     *
     * @return RedirectResponse
     */
    protected function redirect($url, $status = 302)
    {
        return new RedirectResponse($url, $status);
    }

    /**
     * Returns a RedirectResponse to the given route with the given parameters.
     *
     * @param string $route      The name of the route
     * @param array  $parameters An array of parameters
     * @param int    $status     The status code to use for the Response
     *
     * @return RedirectResponse
     */
    protected function redirectToRoute($route, array $parameters = array(), $status = 302)
    {
        return $this->redirect($this->generateUrl($route, $parameters), $status);
    }
}
```

Comment (OskarStark) [inline book/controller.rst]: you are right @hhamon, but to have an example of a status code is not bad imo, what about changing `301` to `302` ??

**Comment by xabbuh** [comment]
Good catch, thanks Hugo.

File: book/controller.rst (hunk 1 above)
Comment (xabbuh) [inline book/controller.rst]: Would you like to open a pull request for that change? By the way, I do not think we need a complete example for that, but we could add a note talking about the fact that redirects by default are responses with a 302 status code but that you can change them through the last parameter of both methods.

Comment (hhamon) [inline book/controller.rst]: Sure! Will do ;)

**Pull request merged**